tests/unit/core/TestScrollToTop.js

```diff
@@ -0,0 +1,58 @@
+import * as React from 'react';
+
+import ScrollToTop, { ScrollToTopBase } from 'core/components/ScrollToTop';
+import {
+  createContextWithFakeRouter,
+  createFakeLocation,
+  shallowUntilTarget,
+} from 'tests/unit/helpers';
+
+describe(__filename, () => {
+  const render = ({ location, ...props } = {}) => {
+    return shallowUntilTarget(<ScrollToTop {...props} />, ScrollToTopBase, {
+      shallowOptions: createContextWithFakeRouter({ location }),
+    });
+  };
+
+  it('renders nothing if there are no children', () => {
+    const root = render();
+    expect(root.html()).toEqual(null);
+  });
+
+  it('renders children', () => {
+    const root = render({ children: <p /> });
+    expect(root.find('p')).toHaveLength(1);
+  });
+
+  it('calls window.scrollTo() when location changes', () => {
+    const _window = {
+      scrollTo: sinon.spy(),
+    };
+
+    const root = render({ _window });
+    sinon.assert.notCalled(_window.scrollTo);
+
+    root.setProps({ location: createFakeLocation() });
+    sinon.assert.calledWith(_window.scrollTo, 0, 0);
+  });
+
+  it('does not call window.scrollTo() when window is not defined', () => {
+    const _window = null;
+
+    const root = render({ _window });
+    root.setProps({ location: createFakeLocation() });
```

Comment:
There isn't an assertion here that proves that `window.scrollTo()` wasn't called. I realize it's hard/impossible to do that, considering that `_window` is `null`, but the test description suggests that it is verifying that `window.scrollTo()` is not called. Do we need this test at all? Or maybe the test should be about making sure an error is not thrown when window is not defined and the location changes?

Reply:
For cases like this I'd typically put a comment at the top of the call, like:

```js
// Make sure this doesn't throw.
root.setProps({ location: createFakeLocation() });
```